### PR TITLE
Correct casing for automatic variables

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Correct casing for special variables (a.k.a. automatic variables)

## PR Context
The following code was used to generate a list of special variables:

```powershell
[psobject].Assembly.GetType('System.Management.Automation.SpecialVariables').GetFields('NonPublic,Static') |
    Where-Object FieldType -eq ([string]) |
    ForEach-Object GetValue $null |
    Where-Object {$_ -imatch '[a-z].+'}
```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
This is a large number of files changed.  Would be easier to review if you can change one variable per commit.

**Comment by xtqqczze** [comment]
@SteveL-MSFT OK, I will rewrite

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 1]
```diff
@@ -185,7 +185,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
 
         It "-File should be able to pass bool string values as string to parameters: <BoolString>" -TestCases @(
             # validates case is preserved
-            @{BoolString = '$truE'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: false positive

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 2]
```diff
@@ -198,7 +198,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
 
         It "-File should be able to pass bool string values as string to positional parameters: <BoolString>" -TestCases @(
             # validates case is preserved
-            @{BoolString = '$tRue'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: false positive

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 3]
```diff
@@ -210,7 +210,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
         }
 
         It "-File should be able to pass bool string values as bool to switches: <BoolString>" -TestCases @(
-            @{BoolString = '$tRue'; BoolValue = 'True'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: false positive

File: test/powershell/engine/ParameterBinding/BooleanParameterDCR.Tests.ps1 [hunk 4]
```diff
@@ -33,7 +33,7 @@ Describe "BooleanParameterDCR Tests" -tags "CI" {
     $tests = @(
             @{ inputTest = $(1 -eq 1); expected = $true },
             @{ inputTest = $true; expected = $true },
-            @{ inputTest = $TRUE; expected = $true }
```
Comment (xtqqczze) [inline test/powershell/engine/ParameterBinding/BooleanParameterDCR.Tests.ps1]: false positive

File: test/powershell/Host/ConsoleHost.Tests.ps1 (hunk 1 above)
Comment (xtqqczze) [reply test/powershell/Host/ConsoleHost.Tests.ps1]: reverted

File: test/powershell/Host/ConsoleHost.Tests.ps1 (hunk 2 above)
Comment (xtqqczze) [reply test/powershell/Host/ConsoleHost.Tests.ps1]: reverted

File: test/powershell/engine/ParameterBinding/BooleanParameterDCR.Tests.ps1 (hunk 4 above)
Comment (xtqqczze) [reply test/powershell/engine/ParameterBinding/BooleanParameterDCR.Tests.ps1]: reverted

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 5]
```diff
@@ -186,7 +186,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
         It "-File should be able to pass bool string values as string to parameters: <BoolString>" -TestCases @(
             # validates case is preserved
             @{BoolString = '$true'},
-            @{BoolString = '$falSe'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: reverted

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 6]
```diff
@@ -199,7 +199,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
         It "-File should be able to pass bool string values as string to positional parameters: <BoolString>" -TestCases @(
             # validates case is preserved
             @{BoolString = '$true'},
-            @{BoolString = '$falSe'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: reverted

File: test/powershell/Host/ConsoleHost.Tests.ps1 [hunk 7]
```diff
@@ -211,7 +211,7 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
 
         It "-File should be able to pass bool string values as bool to switches: <BoolString>" -TestCases @(
             @{BoolString = '$true'; BoolValue = 'True'},
-            @{BoolString = '$faLse'; BoolValue = 'False'},
```
Comment (xtqqczze) [inline test/powershell/Host/ConsoleHost.Tests.ps1]: reverted

File: test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1 [hunk 8]
```diff
@@ -225,7 +225,7 @@ Describe 'Positive Parse Properties Tests' -Tags "CI" {
             [void] m()
             {
                 $LASTEXITCODE
-                $lastexitcode
```
Comment (xtqqczze) [inline test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1]: reverted

File: test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1 [hunk 9]
```diff
@@ -228,7 +228,7 @@ Describe 'Positive Parse Properties Tests' -Tags "CI" {
                 $lastexitcode
                 '111' -Match '1'
                 $Matches
-                $mAtches
```
Comment (xtqqczze) [inline test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1]: reverted

Comment (xtqqczze) [reply test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1]: possible typo here

File: test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1 [hunk 10]
```diff
@@ -231,7 +231,7 @@ Describe 'Positive Parse Properties Tests' -Tags "CI" {
                 $Matches
                 $error[0]
                 $error
-                $pwd
```
Comment (xtqqczze) [inline test/powershell/Language/Classes/Scripting.Classes.BasicParsing.Tests.ps1]: reverted, BasicParsing tests might rely on this behaviour

**Comment by xtqqczze** [comment]
CodeFactor issues are preexisting

**Review by iSazonov — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by xtqqczze** [comment]
@iSazonov why not a squash merge?

**Comment by TravisEz13** [comment]
@iSazonov Yes please use squash commits.  I've remove permissions for merge commits.

**Comment by iSazonov** [comment]
GitHub did return an error after I press `Squash and Merge` button and show something like  "Try again", then ok after refresh. :-(
Merge button was disabled some time ago but then was enabled - I don't know why.

**Comment by xtqqczze** [comment]
Opened issue PowerShell/PSScriptAnalyzer#1401 to detect regressions.

**Pull request merged**